main.py

```diff
@@ -0,0 +1,58 @@
+from sklearn import tree, model_selection
+import graphviz
+from sklearn import ensemble
+from sklearn import preprocessing
+import xgboost as xgb
+from sklearn.model_selection import train_test_split
+import joblib
+import pandas as pd
+
+filename = 'weights/titanic_survivor_grad_boost_classifier.pkl'
+
+
+def convert_csv_to_arrays(path_to_csv: str, train: bool):
+    original_df = pd.read_csv(path_to_csv, sep=',', header=0)\
+        .drop('Ticket', axis=1)\
+        .drop('Cabin', axis=1)
+    original_df.interpolate(inplace=True)
+
+    if train:
+        out_df = original_df['Survived'].copy()
+        in_df = original_df.drop('Survived', axis=1)
+    else:
+        in_df = original_df.copy()
+
+    in_df = in_df
+    in_df['Sex'] = in_df['Sex'].map(lambda x: 1 if x == 'male' else 0)
+    lb = preprocessing.LabelEncoder()
+    min_max_scaler = preprocessing.MaxAbsScaler()
+
+    in_df['Embarked'] = lb.fit_transform(in_df['Embarked'])
+    in_df['Name'] = lb.fit_transform(in_df['Name'])
+    # in_df['Ticket'] = lb.fit_transform(in_df['Ticket'])
+    # in_df['Cabin'] = lb.fit_transform(in_df['Cabin'])
+    in_array = in_df.drop('PassengerId', axis=1).to_numpy()
+    if train:
+        out_array = out_df.to_numpy()
+        return out_array, in_array
+    return in_array, original_df['PassengerId']
+
+
+if __name__ == '__main__':
+    out_array, in_array = convert_csv_to_arrays('data/train.csv', True)
```

Comment:
I suggest using commonly used notation for input features (`X`) and labels (`y`) and in a different order - features first, then labels, i.e.
```suggestion
    X, y = convert_csv_to_arrays('data/train.csv', True)
```